owncloudApp/src/main/java/com/owncloud/android/presentation/ui/sharing/fragments/ShareFileComposeFragment.kt

```diff
@@ -0,0 +1,626 @@
+                            item { EmptyListText(text = stringResource(id = R.string.share_no_public_links)) }
+                        } else {
+                            publicShares = publicShares!!.sortedBy { it.name }
+                            items(publicShares!!) { share ->
+                                SharePublicLinkItem(
+                                    share = share,
+                                    copyOrSend = {
+                                        // Get link from the server and show ShareLinkToDialog
+                                        listener?.copyOrSendPublicLink(share)
+                                    },
+                                    remove = {
+                                        // Remove public link from server
+                                        listener?.showRemoveShare(share)
+                                    },
+                                    edit = {
+                                        listener?.showEditPublicShare(share)
+                                    }
+                                )
+                                Divider(color = colorResource(id = R.color.list_divider_background))
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    override fun onAttach(context: Context) {
+        super.onAttach(context)
+        try {
+            listener = context as ShareFragmentListener?
+        } catch (e: ClassCastException) {
+            throw ClassCastException(activity.toString() + " must implement OnShareFragmentInteractionListener")
+        }
+    }
+
+    override fun onDetach() {
+        super.onDetach()
+        listener = null
+    }
+
+    override fun onResume() {
+        super.onResume()
+        activity?.setTitle(R.string.share_dialog_title)
+    }
+
+    @Composable
+    private fun SharedFileRow() {
+        Row(
+            verticalAlignment = Alignment.CenterVertically,
+            modifier = Modifier.padding(dimensionResource(id = R.dimen.standard_padding))
+        ) {
+            val thumbnail = ThumbnailsCacheManager.getBitmapFromDiskCache(file?.remoteId.toString())
+            if (file!!.isImage && thumbnail != null) {
+                Image(
+                    bitmap = thumbnail.asImageBitmap(),
+                    contentDescription = null,
+                    modifier = Modifier
+                        .size(dimensionResource(id = R.dimen.file_icon_size))
+                        .fillMaxSize()
+                )
+            } else {
+                Image(
+                    painter = painterResource(id = MimetypeIconUtil.getFileTypeIconId(file?.mimetype, file?.fileName)),
+                    contentDescription = null,
+                    modifier = Modifier
+                        .size(dimensionResource(id = R.dimen.file_icon_size))
+                        .fillMaxSize()
```

Comment:
Just curious, why do we set a fixed size and then we set it to fill max size?

Reply:
It seems that that line is not useful at all, so I've removed it 👍